aten/src/ATen/native/cuda/ForeachBinaryOpList.cu

```diff
@@ -0,0 +1,63 @@
+#include <ATen/Dispatch.h>
+#include <ATen/native/ForeachUtils.h>
+#include <ATen/native/cuda/ForeachFunctors.cuh>
+
+namespace at { namespace native {
+
+template<template<class> class Op>
+std::vector<Tensor> foreach_tensor_list_op(TensorList tensors1, TensorList tensors2) {
+    std::vector<std::vector<at::Tensor>> tensor_lists; 
+    std::vector<at::Tensor> vec_res;
+    for (const auto& t: tensors1) {
+        vec_res.emplace_back(at::native::empty_like(t));
+    }
+
+    tensor_lists.emplace_back(std::move(tensors1.vec()));
```

Comment:
This ".vec" operation is very expensive last time I checked. Might be worth doing a for-loop instead of calling into .vec.

Reply:
The expensive part of .vec() is the dynamic allocation involved in creating a new vector. tensors1.vec() creates a new vector. Doing a for-loop to push everything in tensors1 to a new vector also creates a new vector, so they should be the same in terms of performance, right?